forte/data/data_store.py

```diff
@@ -198,14 +202,161 @@ def __init__(
+    def __getstate__(self):
+        r"""
+        In serialization,
+            1) will serialize the annotation sorted list as a normal list;
+            2) will remove `_onto_file_path`, `_dynamically_add_type`, and
+                `entry_dict` to save space.
+        """
+        state = super().__getstate__()
+        for k in state["_DataStore__elements"]:
+            # build the full `_type_attributes`
+            self._get_type_info(k)
+            state["_DataStore__elements"][k] = list(
+                state["_DataStore__elements"][k]
+            )
+        state.pop("_onto_file_path")
+        state.pop("_dynamically_add_type")
+        state.pop("_DataStore__entry_dict")
+        state["entries"] = state.pop("_DataStore__elements")
+        state["fields"] = self._type_attributes
+        for _, v in state["fields"].items():
+            v.pop("parent_entry")
+        return state
+
+    def __setstate__(self, state):
+        r"""
+        In deserialization, we
+            1) transform the annotation list back to a sorted list;
+            2) recreate the `entry_dict` from `__elements`;
+        """
+        super().__setstate__(state)
+        self.__elements = self.__dict__.pop("entries")
+        # add `_type_attributes` to this object to check attributes
+        self._type_attributes = self.__dict__.pop("fields", {})
+        # recreate the `entry_dict` from `__elements`
+        self._DataStore__entry_dict = {}
+
+        for k in self.__elements:
+            if self._is_annotation(k):
+                self.__elements[k] = SortedList(self.__elements[k])
+            for e in self.__elements[k]:
+                # retrieve tid for every entry
+                self._DataStore__entry_dict[e[2]] = e
+
+    @classmethod
+    def deserialize(
+        cls,
+        data_source: str,
+        serialize_method: str = "json",
+        check_attribute: bool = True,
+        silent_mode: bool = True,
+        accept_none: bool = True,
+    ) -> "DataStore":
+        """
+        Deserialize a data_store from a string. This internally calls the
+        internal :meth:`~forte.data.base_store.BaseStore._deserialize` function
+        from :class:`~forte.data.base_store.BaseStore`.
+
+        Args:
+            data_source: The path storing data source.
+            serialize_method: The method used to serialize the data, this
+                should be the same as how serialization is done. The current
+                option is `json`.
+            check_attribute: Boolean value indicating whether users want to
+                check compatibility of attributes. Only applicable when
+                `save_attribute` is set to True in BaseStore.serialize.
+            silent_mode: Boolean value indicating whether users want to
+                see warnings when it checks attributes.
+            accept_none: Boolean value indicating whether users want to
+                fill contradictory fields with none. Only applicable when
```

Comment:
The concept "contradictory fields" is not defined.

Reply:
fixed